### PR TITLE
Fix some minor issues in the English translations

### PR DESCRIPTION
**Pull request by ghost**
This PR accomplishes the following:

1. Javascript has come a long way in the last five years, and `String#startsWith` and `String#endsWith` and `String#includes` exist now. It's a lot cleaner to use them, so we do so.
2. It is possible (albeit unlikely!) for `parenthetical`s to be generated with a generic or rain or sleet type. We are working to reduce the number of cases that this can possibly occur for, but in the meantime we should fix the translation to not be confusing. So: we add a test for this, and clarify that the parenthetical refers to a _chance_ of snow accumulation in those cases. I also updated the README to note this.
3. It was possible for words like "a"/"of"/"with"/etc. to get capitalized. This shouldn't happen in English.
4. Did a bunch of cleanup and added comments.

### CODE REVIEW

**Review by hhansson — APPROVED** [review]
very satisfying.

File: test_cases/en.json [hunk 1]
```diff
@@ -260,9 +263,11 @@
       ["starting-in", "very-light-rain", ["less-than", ["minutes", 1]]]],
 
   "Next hour forecasts are temporarily unavailable due to all nearby radar stations being offline.":
-  ["sentence",["next-hour-forecast-status", "temporarily-unavailable", "station-offline"]],
+    ["sentence", ["next-hour-forecast-status", "temporarily-unavailable", "station-offline"]],
```
Comment (nonrenewable) [inline test_cases/en.json]: we should make the linter yell about these since otherwise i'll keep doing it 😞

**Review by nonrenewable-zz — APPROVED** [review]
it's def pretty! thank you for atoning for my sins...

File: test_cases/en.json (hunk 1 above)
Comment (ironwallaby) [reply test_cases/en.json]: The linter doesn't check JSON :(

```
midori:translations jay$ ./node_modules/.bin/eslint test_cases/en.json 

/Users/jay/Projects/Dark Sky/translations/test_cases/en.json
  2:10  error  Parsing error: Unexpected token :

✖ 1 problem (1 error, 0 warnings)

```

Comment (nonrenewable) [reply test_cases/en.json]: 😭

**Pull request merged**